include/boost/gil/concepts/pixel_iterator.hpp

```diff
@@ -45,48 +43,48 @@ template <typename It> struct iterator_adaptor_get_base;
 // Isolating mutability to result in faster compile time
 namespace detail {
 
-// Preconditions: TT Models boost_concepts::ForwardTraversalConcept
-template <class TT>
-struct ForwardIteratorIsMutableConcept
-{
-    void constraints()
-    {
-        auto const tmp = *i;
-        *i++ = tmp; // require postincrement and assignment
-    }
-    TT i;
-};
-
-// Preconditions: TT Models boost::BidirectionalIteratorConcept
-template <class TT>
-struct BidirectionalIteratorIsMutableConcept
-{
-    void constraints()
-    {
-        gil_function_requires< ForwardIteratorIsMutableConcept<TT>>();
-        auto const tmp = *i;
-        *i-- = tmp; // require postdecrement and assignment
-    }
-    TT i;
-};
-
-// Preconditions: TT Models boost_concepts::RandomAccessTraversalConcept
-template <class TT>
-struct RandomAccessIteratorIsMutableConcept
-{
-    void constraints()
-    {
-        gil_function_requires<BidirectionalIteratorIsMutableConcept<TT>>();
-
-        typename std::iterator_traits<TT>::difference_type n = 0;
-        ignore_unused_variable_warning(n);
-        i[n] = *i; // require element access and assignment
-    }
-    TT i;
-};
+//// Preconditions: TT Models ForwardIterator
+//template <class TT>
+//struct ForwardIteratorIsMutableConcept
+//{
+//    void constraints()
```

Comment:
I dream about switching over to C++20 concepts :-)

Reply:
Baby steps... But I agree that moving to C++20, at least for concept checking, would be ideal.